lib/ToolbarBadgeHub.jsm

```diff
@@ -86,12 +103,19 @@ class _ToolbarBadgeHub {
     }
   }
 
-  executeAction({ id }) {
+  executeAction({ id, data }) {
     switch (id) {
       case "show-whatsnew-button":
         ToolbarPanelHub.enableToolbarButton();
         ToolbarPanelHub.enableAppmenuButton();
         break;
+      case "moments-wnp":
+        const { url, expire } = data;
+        Services.prefs.setStringPref(
+          HOMEPAGE_OVERRIDE_PREF,
```

Comment:
Not sure if we should only set if pref isn't already set (i.e., a previous pending `once` / `moments-wnp` action)? Could be additional `targeting` condition or something all `moments-wnp` automatically check or … just allow overriding an existing one?

Reply:
Good question. Maybe we should always write if the message is selected and control this via targeting. This way we could ship an `undo` message by setting targeting to `true`. What do you think?

Reply:
Overwriting in some cases definitely makes sense, like if you decide to remove the message from remote settings or add a more important one.